build/templates/_converters.py.mako

```diff
@@ -0,0 +1,32 @@
+<%
+    config = template_parameters['metadata'].config
+    module_name = config['module_name']
+%>\
+from ${module_name} import visatype
+
+
+def _timedelta_converter(value, library_type, scaling):
+    if str(type(value)).find("'datetime.timedelta'") != -1:
```

Comment:
What is this if statement for? Is it for backwards compatilibilty / allow users to pass a number in the public API?
If that's the case, I think we should remove it. Does that make a mess of things with defaults?

Reply:
I believe we should support a bare number as well as a timedelta object.

Removing it is what makes a mess of defaults.
* Right now we only import datetime if we need it
  * We would have to import it before the function so we could do something like `timeout=datetime.timedelta(seconds=5.0)`
* We would have to add support for doing something like `timeout=datetime.timedelta(seconds=5.0)`
  * The code generator as is I don't think would support that.

Reply:
> Right now we only import datetime if we need it

We think all the APIs have at least one function with a timeout, so we could add the import to session.py.mako and be done with it.

> The code generator as is I don't think would support that.
 
You're right. This is a bummer. I tried it and it generates (note extra quotes):

     def wait_for_event(self, event_id, timeout='datetime.timedelta(10.0)'):

It seems we added logic to the generator to handle enums. Maybe we got too clever and we make it so that the default must be specified as a string (not a number) in the metadata and is copied verbatim (for strings we'd have to make sure the string literal includes quotes).

I don't think this is too hard to fix and may even be a simplification. It's weird that some APIs take in numbers as float, others as int, different units... in addition to timedelta.

From [Zen of Python](https://www.python.org/dev/peps/pep-0020/#id3)
_"There should be one-- and preferably only one --obvious way to do it."_

Allowing numbers breaks this.